component/src/test/java/io/siddhi/extension/io/cdc/source/TestCaseOfCDCListeningModeMongo.java

```diff
@@ -0,0 +1,304 @@
+    private String collectionName = "SweetProductionTable";
+
+    @BeforeClass
+    public void initializeConnectionParams() {
+        databaseUri = "mongodb://<host>:<port>/<collection_name>";
+        replicaSetUri = "jdbc:mongodb://<replica_set_name>/<host>:<port>/<database_name>";
+        username = "user_name";
+        password = "password";
+    }
+
+    @BeforeMethod
+    public void init() {
+        eventCount.set(0);
+        eventArrived.set(false);
+        currentEvent = new Event();
+    }
+
+    /**
+     * Test case to Capture Insert operations from a Mongo table.
+     */
+    @Test
+    public void testInsertCDC() throws InterruptedException {
+        log.info("------------------------------------------------------------------------------------------------");
+        log.info("CDC TestCase: Capturing Insert change data from Mongo.");
+        log.info("------------------------------------------------------------------------------------------------");
+
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String cdcinStreamDefinition = "@app:name('cdcTesting')" +
+                "@source(type = 'cdc'," +
+                " url = '" + replicaSetUri + "'," +
+                " username = '" + username + "'," +
+                " password = '" + password + "'," +
+                " table.name = '" + collectionName + "', " +
+                " operation = 'insert', " +
+                " @map(type='keyvalue'))" +
+                "define stream istm (id string, name string, amount double, volume int);";
+
+        String mongoStoreDefinition = "define stream insertionStream (name string, amount double, volume int);" +
+                "@Store(type='mongodb', mongodb.uri='" + databaseUri + "')" +
+                "define table SweetProductionTable (name string, amount double, volume int);";
+
+        String mongoQuery = "@info(name='query2') " +
+                "from insertionStream " +
+                "insert into SweetProductionTable;";
+
+        SiddhiAppRuntime cdcAppRuntime = siddhiManager.createSiddhiAppRuntime(cdcinStreamDefinition);
+
+        StreamCallback insertionStreamCallback = new StreamCallback() {
+            @Override
+            public void receive(Event[] events) {
+                for (Event event : events) {
+                    currentEvent = event;
+                    eventCount.getAndIncrement();
+                    log.info(eventCount + ". " + event);
+                    eventArrived.set(true);
+                }
+            }
+        };
+
+        cdcAppRuntime.addCallback("istm", insertionStreamCallback);
+        cdcAppRuntime.start();
+
+        Thread.sleep(1000);
+
+        QueryCallback mongoQueryCallback = new QueryCallback() {
+            @Override
+            public void receive(long timestamp, Event[] inEvents, Event[] removeEvents) {
+                for (Event event : inEvents) {
+                    log.info("insert done: " + event);
```

Comment:
Can't we add an assert here, instead of just logging?

Reply:
Assertion added for the event received at the CDC source.